network/Message.cpp

```diff
@@ -85,15 +94,74 @@ char* Message::Data() noexcept
 void Message::Swap(Message& rhs) noexcept {
     std::swap(m_type, rhs.m_type);
     std::swap(m_message_size, rhs.m_message_size);
+    std::swap(m_is_compressed, rhs.m_is_compressed);
     std::swap(m_message_text, rhs.m_message_text);
 }
 
 void Message::Reset() noexcept {
     m_type = MessageType::UNDEFINED;
     m_message_size = 0;
+    m_is_compressed = false;
     m_message_text.clear();
 }
 
+void Message::Compress() noexcept {
+    if (!m_is_compressed) {
+        std::string_view::size_type temp = m_message_size;
+        m_message_text = CompressImpl(m_message_text);
+        m_message_size = m_message_text.size();
```

Comment:
could perhaps leave this as the uncompressed size, so that it can be used to allocate space when decompressing

Reply:
We have to send compressed size as well so network part will know where message ends.

Reply:
In https://github.com/freeorion/freeorion/pull/3854/commits/4ead6795ce2b51bce839955a679750840ada4831:
- Header expanded - again - to include the expected uncompressed size of the data.
- `m_message_size` is used to store the uncompressed size, and `m_message_cpr_size` is used to store the compressed size.
- `Compress()` updates `m_message_cpr_size`